Source/WarpX.H

```diff
@@ -41,6 +41,7 @@
 #include "FieldSolver/Fields.H"
 #include "FieldSolver/ElectrostaticSolver.H"
 #include "FieldSolver/MagnetostaticSolver/MagnetostaticSolver.H"
+#include "FieldSolver/ImplicitSolvers/ImplicitSolverLibrary.H"
```

Comment:
In order to speed up compiles that use `WarpX.H`, I think technically here we only need
```suggestion
#include "FieldSolver/ImplicitSolvers/WarpXSolverVec.H"
```
and
```C++
#include "FieldSolver/ImplicitSolvers/ImplicitSolverLibrary.H"
```
is only needed in `WarpX.cpp`, right? :)

Reply:
I made this change.

Since ImplicitSolver.H is not in WarpX.H (which is passed around a lot), is there still a need to do a forward declaration?